code/datums/elements/tool_renaming.dm

```diff
@@ -0,0 +1,68 @@
+/**
+ * Renaming tool element
+ *
+ * When using this tool on an object with UNIQUE_RENAME,
+ * lets the user rename/redesc it.
+ */
+/datum/element/tool_renaming
+
+/datum/element/tool_renaming/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ITEM_INTERACTING_WITH_ATOM, PROC_REF(attempt_rename))
+
+/datum/element/tool_renaming/Detach(datum/source)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_INTERACTING_WITH_ATOM)
+
+/datum/element/tool_renaming/proc/attempt_rename(datum/source, mob/living/user, atom/interacting_with, list/modifiers)
+	if(!isobj(interacting_with))
+		return NONE
+	
+	var/obj/renamed_obj = interacting_with
+
+	if(!(renamed_obj.obj_flags & UNIQUE_RENAME))
+		return NONE
```

Comment:
```suggestion
	if(!(renamed_obj.obj_flags & UNIQUE_RENAME))
		balloon_alert(user, "can't rename this!")
		return NONE
```

Reply:
No balloon alert. Otherwise it will pop this up every time you click on something with a pen.

Reply:
pretty much yeah